flow.go

```diff
@@ -47,6 +51,15 @@ func (i Identifier) String() string {
 	return i.Hex()
 }
 
+func (i Identifier) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("\"%s\"", i.Hex())), nil
+}
+
+func (i *Identifier) UnmarshalJSON(data []byte) error {
+	*i = HexToID(strings.Trim(string(data), "\""))
+	return nil
```

Comment:
I gave this some thought and I think I agree with you.

It's tricky because a zeroed array has two meanings in Go -- either the value was never initialized (empty) or the ID actually is `0x000...` (which is technically a possible hash value).

We _could_ use `*flow.Identifier`, which would allow us to map `nil` => `null`, but I'd prefer not to do that.

I think the only _safe_ thing to do is marshal a zeroed IDs and addresses as `0x000...`.